functions/export-csv.js

```diff
@@ -21,13 +21,61 @@ const {db} = require('./common/context');
 
 function exportCsv(req, res) {
   const {
-    project: providedProjectId,
-    featureType: providedFeatureTypeId,
+    project: projectId,
+    featureType: featureTypeId,
     lang: desiredLanguage,
   } = req.query;
 
-  return res.status(500).send('Not supported yet.');
+  let data = {};
+  return db.fetchProject(projectId).then(
+    project => {
+      if (!project.exists) {
+        res.status(404).send('not found');
+        return Promise.reject(new Error('project not found: ' + projectId))
+      }
+      res.type('text/csv');
+      let form = project.get('featureTypes')[featureTypeId]['forms'];
+      let elements = form[Object.keys(form)[0]]['elements'];
+      var arr = []
+      for (var el in elements) {
+        arr.push('"' + escapeQuotes(elements[el]['labels']['_']) + '"');
+      }
+      res.write(arr.join() + '\n');
+      
+      return Promise.all(
```

Comment:
You shouldn't need to wrap these in `Promise.all`; you can just call `db.fetchRecords(..).then( <logic below > )`

Reply:
I need to send 'elements' variable as well, hence using Promise.all.

Reply:
You should already have access to `elements` in this scope, and sub-scopes (i.e., in the lambda provided to the Promise), no?

Reply:
I created the elements variable in the first promise, it was not in the scope of chained promise. But I can declare it before promises so that it'll be visible without passing, will do that.

Reply:
I probably don't understand how `const` and `let` scopes work in ES6 then.. I'll read up a bit. In the meantime perhaps add a TODO to clean this up later?

Reply:
Sure!